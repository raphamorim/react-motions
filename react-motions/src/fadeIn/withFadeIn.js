import React from 'react'
import styles from './styles'

function withFadeIn(Component) {
  const style = {
    WebkitAnimationName: "fadeIn",
    animationName: "fadeIn",
    WebkitAnimationDuration: "1s",
    animationDuration: "1s",
    WebkitAnimationFillMode: "both",
    animationFillMode: "both"
  }

  return (
    <div style={style}>
      {Component}
    </div>
  )
}

export default withFadeIn
