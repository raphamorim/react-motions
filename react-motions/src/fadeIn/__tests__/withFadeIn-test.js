import React from 'react';
import withFadeIn from '../withFadeIn';

describe('withFadeIn', () => {
  it('should render', () => {
    const wrapper = shallow(withFadeIn(<div>FadeIn</div>));

    expect(wrapper).toMatchSnapshot();
  });
});
