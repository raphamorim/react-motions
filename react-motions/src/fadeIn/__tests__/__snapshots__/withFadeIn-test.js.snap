// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFadeIn should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    style={
      Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "fadeIn",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "fadeIn",
      }
    }
  >
    <div>
      FadeIn
    </div>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        FadeIn
      </div>,
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "fadeIn",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "fadeIn",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "FadeIn",
      },
      "ref": null,
      "rendered": "FadeIn",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          FadeIn
        </div>,
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "fadeIn",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "fadeIn",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FadeIn",
        },
        "ref": null,
        "rendered": "FadeIn",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
